x-pack/plugins/enterprise_search/public/applications/app_search/components/role_mappings/role_mappings_logic.test.ts

```diff
@@ -0,0 +1,437 @@
+        RoleMappingsLogic.actions.setRoleMappingData(mappingServerProps);
+
+        expect(RoleMappingsLogic.values.roleMapping).toEqual(asRoleMapping);
+        expect(RoleMappingsLogic.values.dataLoading).toEqual(false);
+        expect(RoleMappingsLogic.values.attributes).toEqual(mappingServerProps.attributes);
+        expect(RoleMappingsLogic.values.availableEngines).toEqual(
+          mappingServerProps.availableEngines
+        );
+        expect(RoleMappingsLogic.values.accessAllEngines).toEqual(true);
+        expect(RoleMappingsLogic.values.elasticsearchRoles).toEqual(
+          mappingServerProps.elasticsearchRoles
+        );
+        expect(RoleMappingsLogic.values.selectedEngines).toEqual(
+          new Set(engines.map((e) => e.name))
+        );
+      });
+
+      it('sets default group with new role mapping', () => {
+        RoleMappingsLogic.actions.setRoleMappingData({
+          ...mappingServerProps,
+          roleMapping: undefined,
+        });
+
+        expect(RoleMappingsLogic.values.selectedEngines).toEqual(new Set());
+      });
+    });
+
+    it('handleRoleChange', () => {
+      RoleMappingsLogic.actions.handleRoleChange('dev');
+
+      expect(RoleMappingsLogic.values.roleType).toEqual('dev');
+    });
+
+    it('handleEngineSelectionChange', () => {
+      const engine = engines[0];
+      const otherEngine = engines[1];
+      RoleMappingsLogic.actions.setRoleMappingData({
+        ...mappingServerProps,
+        roleMapping: {
+          ...asRoleMapping,
+          engines: [engine, otherEngine],
+        },
+      });
+
+      RoleMappingsLogic.actions.handleEngineSelectionChange(otherEngine.name, true);
+      expect(RoleMappingsLogic.values.selectedEngines).toEqual(
+        new Set([engine.name, otherEngine.name])
+      );
+
+      RoleMappingsLogic.actions.handleEngineSelectionChange(otherEngine.name, false);
+      expect(RoleMappingsLogic.values.selectedEngines).toEqual(new Set([engine.name]));
+    });
+
+    it('handleAccessAllEnginesChange', () => {
+      RoleMappingsLogic.actions.handleAccessAllEnginesChange();
+
+      expect(RoleMappingsLogic.values.accessAllEngines).toEqual(false);
+    });
+
+    describe('handleAttributeSelectorChange', () => {
+      const elasticsearchRoles = ['foo', 'bar'];
+
+      it('sets values correctly', () => {
+        RoleMappingsLogic.actions.setRoleMappingData({
+          ...mappingServerProps,
+          elasticsearchRoles,
+        });
+        RoleMappingsLogic.actions.handleAttributeSelectorChange('role', elasticsearchRoles[0]);
+
+        expect(RoleMappingsLogic.values.attributeValue).toEqual(elasticsearchRoles[0]);
+        expect(RoleMappingsLogic.values.attributeName).toEqual('role');
+      });
+
+      it('correctly handles "role" fallback', () => {
+        RoleMappingsLogic.actions.handleAttributeSelectorChange('username', elasticsearchRoles[0]);
+
+        expect(RoleMappingsLogic.values.attributeValue).toEqual('');
+      });
```

Comment:
General comment, even if you don't necessarily *need* to use `mount` to set up test data, it is often nice to do so for 2 reasons:

1. It makes a better before and after diff, as mentioned in my previous comment.
2. You're ensuring what you meant to change is *actually* changing.

Meaning, if `attributeValue` has a default value of `''` before this test was run, then this function could do absolutely nothing and the test would still pass.


```suggestion
      it('correctly handles "role" fallback', () => {
        mount({
          attributeValue: 'foo',
        });
        
        RoleMappingsLogic.actions.handleAttributeSelectorChange('username', elasticsearchRoles[0]);

        expect(RoleMappingsLogic.values.attributeValue).toEqual('');
      });
```

Also, this probably should test `attributeValue` as well, right?

This is the last comment I'll make about descriptions, I think this comment applies to many of these tests. But again, I think a lot of these test could use better descriptions. I'm not actually sure what this description means.

Reply:
Can you elaborate on this statement?
> Also, this probably should test attributeValue as well, right?

Reply:
I was just thinking that attributeValue gets changed in this method as well... but maybe it doesn't 🤔 . Feel free to ignore.

Reply:
Addressed here (https://github.com/elastic/kibana/pull/94461/commits/eb1298a3bd9c43e9251d110e70c290dce6be9966)